packages/ai/src/ai/modules/task/task_conn.py

```
@@ -164,10 +166,24 @@ def __init__(
+        self._connected_at: float = time.time()
+        self._messages_in: int = 0
+        self._messages_out: int = 0
+        self._last_activity: float = time.time()
+        self._client_info: Dict[str, str] = {}
+
+    async def send(self, message: Dict[str, Any]) -> None:
+        self._messages_out += 1
+        self._last_activity = time.time()
+        await super().send(message)
+
     async def on_receive(self, message: Dict[str, Any] = {}) -> None:
```

Ruff (0.15.7)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)